core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SharedHttpClientSessionManager.java

```diff
@@ -314,17 +473,67 @@ protected final ExecutorService getExecutorService() {
 	}
 
 	private CloseableHttpClient createHttpClient() {
+
 		HttpClientBuilder nextHttpClientBuilder = httpClientBuilder;
 		if (nextHttpClientBuilder != null) {
 			return nextHttpClientBuilder.build();
 		}
 
+		RequestConfig requestConfig = getDefaultRequestConfig();
+
 		return HttpClientBuilder.create()
 				.evictExpiredConnections()
+				.evictIdleConnections(30, TimeUnit.MINUTES)
 				.setRetryHandler(retryHandlerStale)
 				.setServiceUnavailableRetryStrategy(serviceUnavailableRetryHandler)
 				.useSystemProperties()
-				.setDefaultRequestConfig(RequestConfig.custom().setCookieSpec(CookieSpecs.STANDARD).build())
+				.setDefaultRequestConfig(requestConfig)
```

Comment:
we are also using a configurable value for the max number of connections

```
.setMaxConnPerRoute(maxConnections)
.setMaxConnTotal(maxConnections)
```

Reply:
That's a very good idea. What are your defaults?

I think that part of the issue is simply that the default number of connections is too low, maybe even 1, which is quite terrible for the way that RDF4J typically does joins.

Reply:
We have currently 25 connections as default value (for quite some time). Note that we haven't done systematic benchmarks on the impact of adjusting the number of connections.